WordPressVIPMinimum/Tests/Security/ProperEscapingFunctionUnitTest.inc

```diff
@@ -30,12 +30,44 @@ echo "<a title='" . esc_html( $some_var ) . "'></a>"; // NOK.
 
 echo '<media:content url="' . esc_url( $post_image ) . '" medium="image">'; // OK.
 
-echo '<media:content url="' . esc_attr( $post_image ) . '" medium="image">'; // NOK.
+echo '<media:content url="' . esc_attr( $post_image ) . '" medium="image">'; // Error.
 
 echo 'data-param-url="' . esc_url( $share_url ) . '"'; // OK.
 
-echo 'data-param-url="' . esc_html( $share_url ) . '"'; // NOK.
+echo 'data-param-url="' . esc_html( $share_url ) . '"'; // Error.
 
 ?>
 
 <form method="post" action="<?php echo esc_html(admin_url('admin.php?page='.$base_name.'&amp;mode=logs&amp;id='.$poll_id)); ?>">
+
+?>
+
+<a href="#link"><?php echo esc_attr( 'testing' ); // Error.
+?> </a>
+
+<?php echo ' <h' . "2>    " ,        esc_attr( $heading ) . "</h2">; // Error.
+?> <style>
+h1 {
+  color: <?php echo esc_attr( $color ) ?>; /* OK */
+  font-family: Arial;
+  font-size: 10px;
+}
+</style>
+
+<article id="foo-bar" data-property="<?php echo esc_attr( $year ); // OK.
+?>">
+Test
+</article>
+
+<h1><?php echo esc_attr__( $title, 'domain' ); ?></h1> <!-- Error --> ?>
+<?php echo '<h1>' . esc_attr__( $some_var, 'domain' ) . '</h1>'; // Error.
+echo '<h1>',      esc_attr__( $title, 'domain' ), '</h1>'; // Error.
+echo "<" . "h1>" . esc_attr( $test ) . '</h1>'; // Error.
+?>
+<h1> <?php echo esc_attr( $title ) . '</h1>'; ?> // Error.
+<div><?= esc_attr($attr) ?></div><!-- Error -->
+<div><?php esc_attr_e($attr) ?></div><!-- Error -->
+<div data-tag="<?= esc_attr( $attr ); ?>"> <!-- OK -->
+<?php echo "<div> {$test} </div>"; // OK.
+echo "<{$tag}>" . esc_attr( $tag_content ) . "</{$tag}>"; // Error.
+echo "<$tag" . '   >' . esc_attr( $tag_content ) . "</$tag>"; // Error.
```

Comment:
What is this line testing ?

Reply:
This line is testing that there can be a concatenated start tag before `esc_attr()` is called will still flag an error.

Reply:
It doesn't test anything new but it ensures that this use case will not be broken if the sniff is changed in the future.